packages/flutter/test/material/navigation_bar_test.dart

```diff
@@ -0,0 +1,429 @@
+      ),
+    );
+  });
+
+  testWidgets('Navigation bar semantics with some labels hidden', (WidgetTester tester) async {
+    final Key painterKey = UniqueKey();
+    Widget _widget({int selectedIndex = 0}) {
+      return _buildWidget(
+        painterKey,
+        NavigationBar(
+          labelBehavior: NavigationDestinationLabelBehavior.onlyShowSelected,
+          selectedIndex: selectedIndex,
+          destinations: const <Widget>[
+            NavigationDestination(
+              icon: Icon(Icons.ac_unit),
+              label: 'AC',
+            ),
+            NavigationDestination(
+              icon: Icon(Icons.access_alarm),
+              label: 'Alarm',
+            ),
+          ],
+        ),
+      );
+    }
+
+    await tester.pumpWidget(_widget(selectedIndex: 0));
+
+    expect(
+      tester.getSemantics(find.text('AC')),
+      matchesSemantics(
+        label: 'AC\nTab 1 of 2',
+        textDirection: TextDirection.ltr,
+        isFocusable: true,
+        isSelected: true,
+        hasTapAction: true,
+      ),
+    );
+    expect(
+      tester.getSemantics(find.text('Alarm')),
+      matchesSemantics(
+        label: 'Alarm\nTab 2 of 2',
+        textDirection: TextDirection.ltr,
+        isFocusable: true,
+        isSelected: false,
+        hasTapAction: true,
+      ),
+    );
+
+    await tester.pumpWidget(_widget(selectedIndex: 1));
+
+    expect(
+      tester.getSemantics(find.text('AC')),
+      matchesSemantics(
+        label: 'AC\nTab 1 of 2',
+        textDirection: TextDirection.ltr,
+        isFocusable: true,
+        isSelected: false,
+        hasTapAction: true,
+      ),
+    );
+    expect(
+      tester.getSemantics(find.text('Alarm')),
+      matchesSemantics(
+        label: 'Alarm\nTab 2 of 2',
+        textDirection: TextDirection.ltr,
+        isFocusable: true,
+        isSelected: true,
+        hasTapAction: true,
+      ),
+    );
+  });
+
+  testWidgets('Navigation bar does not grow with text scale factor', (WidgetTester tester) async {
+    final Key painterKey = UniqueKey();
+    const int _animationMilliseconds = 800;
+
+    Widget _widget({double textScaleFactor = 1}) {
+      return _buildWidget(
+        painterKey,
+        MediaQuery(
+          data: MediaQueryData(textScaleFactor: textScaleFactor),
+          child: NavigationBar(
+            animationDuration: const Duration(milliseconds: _animationMilliseconds),
+            destinations: const <NavigationDestination>[
+              NavigationDestination(
+                icon: Icon(Icons.ac_unit),
+                label: 'AC',
+              ),
+              NavigationDestination(
+                icon: Icon(Icons.access_alarm),
+                label: 'Alarm',
+              ),
+            ],
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(_widget());
+    final double initialHeight =
+        tester.renderObject<RenderBox>(find.byType(NavigationBar)).size.height;
```

Comment:
Why not use one line and [getSize](https://api.flutter.dev/flutter/flutter_test/WidgetController/getSize.html) (here and below)
```dart
tester.getSize(find.byType(NavigationBar)).height
```

Reply:
Done